lib/super_good/solidus_taxjar/reportable.rb

```diff
@@ -46,14 +46,19 @@ def order_reportable?(order)
           order.shipped?
       end
 
-      # @return [Boolean] true if the transaction has been previously reported
-      #   to TaxJar, the order is currently in `paid` state and there is a
-      #   difference between the total (before tax) on the order in Solidus
-      #   and the transaction amount on TaxJar.
+      # @return [Boolean] true if the order is currently in `paid` state and
+      #   if the transaction has been previously reported to TaxJax checks
+      #   for a difference between the total (before tax) on the order in Solidus
+      #   and the transaction amount on TaxJar. If the transaction has not been
+      #   previously reported, it returns true as long as it's in `paid` state.
       def transaction_replaceable?(order)
-        order.taxjar_order_transactions.present? &&
-          order.payment_state == "paid" &&
+        return unless order.payment_state == "paid"
+
+        if order.taxjar_order_transactions.present?
           amount_changed?(order)
+        else
+          true
+        end
```

Comment:
Looks like you came to the same conclusion based on your recently comments on the PR. But I think we should not merge this change as-is. With this change, reporting and replacing isn't meaningfully different, so we should just merge them into a single function.

Reply:
Just wanted to highlight this change for further discussion if necessary.